src/synapse/azext_synapse/commands.py

```diff
@@ -42,6 +53,9 @@ def load_command_table(self, _):
         g.custom_command('list', 'list_workspaces')
         g.custom_command('create', 'create_workspace', supports_no_wait=True)
         g.custom_command('update', 'update_workspace', supports_no_wait=True)
+        g.custom_command('check-name', 'custom_check_name_availability',
+                         command_type=synapse_operations_sdk,
+                         client_factory=cf_synapse_client_operations_factory)
         g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
         g.wait_command('wait')
 

```

Comment:
For `az synapse spark pool list`, we should support `--ids`. The resource id should target the resource id of some workspace. Reference: `az sql db list --ids`. Same comments with `az synapse sql pool list`

Reply:
> For `az synapse spark pool list`, we should support `--ids`. The resource id should target the resource id of some workspace. Reference: `az sql db list --ids`. Same comments with `az synapse sql pool list`

This comment is conflict with cli document. But anyway I have supported this feature.

Reply:
Could ypu please share the doc link?

Reply:
[https://github.com/Azure/azure-cli/blob/b1db5722823d2d00d5827eb6041c5b73f31bfae0/doc/authoring_command_modules/authoring_commands.md#supporting-the-ids-parameter](https://github.com/Azure/azure-cli/blob/b1db5722823d2d00d5827eb6041c5b73f31bfae0/doc/authoring_command_modules/authoring_commands.md#supporting-the-ids-parameter)

"--ids is intended to be the ID of the resource the command group is about. Thus, it needs to be suppressed on list commands for child resources. "

---
In reply to: [416230837](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r416230837) [](ancestors = 416230837)